main.tf

```
@@ -386,7 +386,7 @@ resource "aws_eks_addon" "this" {
   cluster_name = aws_eks_cluster.this[0].name
   addon_name   = try(each.value.name, each.key)
 
-  addon_version            = try(each.value.addon_version, data.aws_eks_addon_version.this[each.key].version)
+  addon_version            = coalesce(each.value.addon_version, data.aws_eks_addon_version.this[each.key].version)
```

we have to keep the try on the map lookup; coalesce will not handle the case where the lookup does not return the key
```suggestion
  addon_version            = coalesce(try(each.value.addon_version, null), data.aws_eks_addon_version.this[each.key].version))
```